p2p/discovery/database.go

```diff
@@ -0,0 +1,52 @@
+/**
+*  @file
+*  @copyright defined in go-seele/LICENSE
+ */
+package discovery
+
+import (
+	"github.com/seeleteam/go-seele/common"
+	"sync"
+)
+
+type database struct {
+	m map[common.Hash]*Node // TODO use memory for temp, will use level db later
+
+	mutex sync.Mutex
+}
+
+func NewDatabase() *database {
+	return &database{
+		m: make(map[common.Hash]*Node),
+	}
+}
+
+func (db *database) add(id common.Hash, value *Node)  {
+	db.mutex.Lock()
+	defer db.mutex.Unlock()
```

Comment:
so many "db.mutex.Lock; defer db.mutex.Unlock()", you'd better to extract a common func to make the code more clear.

Reply:
I think this is how to use lock :)